src/Amazon.AspNetCore.DataProtection.SSM/PersistOptions.cs

```diff
@@ -28,5 +29,11 @@ public class PersistOptions
         /// don't specify a key ID, the system uses the default key associated with your AWS account.
         /// </summary>
         public string KMSKeyId { get; set; }
+
+        /// <summary>
+        /// Whether the store is allowed to create SNN parameters using the Advanced Tier pricing in case that the stored key is to big for standard tier.
+        /// Standard tier maximum size is 4096 characters (4KB).
+        /// </summary>
+        public bool CanUseAdvancedTier { get; set; }
```

Comment:
I think it would be better as an Enum than a boolean. I can imagine some users always wanting to use Advanced Tier. Some don't care and want the library to use whatever best fits, and some users only want to use standard. By making this enum it gives us room to make changes in the future. Also rename the property to something like `TierStorageMode`.

Reply:
Agreed 👍